tutorials/onchain-governance/fast-track-gov-proposal.md

```diff
@@ -0,0 +1,224 @@
+
+```typescript hl_lines="2-3"
+--8<-- 'code/tutorials/onchain-governance/fast-track-gov-proposal/test-proposal.ts:226:229'
+  ...
+}
+```
+
+### Create and Submit the Proposal
+
+Create a `generateProposal` function responsible for preparing and submitting the on-chain proposal. The function implements a workflow that involves the following:
+
+1. Configuring a keyring for the Alice account to facilitate transaction submission. When using Chopsticks, this development account is pre-funded to execute all necessary actions
+
+2. Executing a batch transaction that comprises three critical operations:
+    1. **`preimage.notePreimage`** - registers a [preimage](/polkadot-protocol/glossary#preimage){target=\_blank} using the selected call
+
+        !!!note
+            The preimage hash is simply the hash of the proposal to be enacted. The on-chain proposals do not require the entire image of extrinsics and data (for instance, the Wasm code, in case of upgrades) to be submitted but would need that image's hash. That preimage can be submitted and stored on-chain against the hash later upon the proposal's dispatch.
+
+    2. **`referenda.submit`** - submits the proposal to the referenda system. It uses the preimage hash extracted from the call as part of the proposal submission process. The proposal is submitted with the selected origin
+
+    3. **`referenda.placeDecisionDeposit`** - places the required decision deposit for the referendum. This deposit is required to move the referendum from the preparing phase to the deciding phase
+
+```typescript
+--8<-- 'code/tutorials/onchain-governance/fast-track-gov-proposal/test-proposal.ts:generateProposal'
+```
+
+Within the `main` function, define the specific call you want to execute and its corresponding origin, then invoke the `generateProposal` method:
+
+```typescript hl_lines="5-14"
+--8<-- 'code/tutorials/onchain-governance/fast-track-gov-proposal/test-proposal.ts:226:240'
+  ...
+}
+```
+
+!!!note
+    The [`setCodeWithoutChecks`](https://paritytech.github.io/polkadot-sdk/master/frame_system/pallet/struct.Pallet.html#method.set_code_without_checks){target=\_blank} extrinsic used in this example is for demonstration purposes only. Replace it with the specific extrinsic that matches your governance proposal's intended functionality. Ensure the call matches your target Polkadot SDK-based network's runtime requirements and governance process.
+
+### Force Proposal Execution
+
+After submitting your proposal, you can test its execution without waiting for the standard voting and enactment periods. Chopsticks provides a way to force the execution of a proposal by directly manipulating the chain state and scheduler.
+
+Create a utility function called `moveScheduledCallTo` and copy the following code. This function helps move a scheduled call to a specific future block in a forked chain by matching certain criteria.
```

Comment:
There's a lot that happens in the `moveScheduledCallTo` function and this is the only two sentences about it. Not that I think there should be paragraphs about it, but we should really be instructing users on what to do inside of these functions instead of just being like ok, here you go. Because the goal of a tutorial is to learn by building and understanding each step along the way instead of skipping over the steps entirely